app/synchronizationerror.cpp

```diff
@@ -37,6 +37,11 @@ SynchronizationError::ErrorType SynchronizationError::errorType( int errorCode,
     // Project no longer exists / is on different server
     return ErrorType::ProjectNotFound;
   }
+  else if ( errorCode == 422 )
+  {
+    // Hit the maximun number of contributor per month
+    return ErrorType::MonthlyContributorsLimitHit;
+  }
```

Comment:
`422` error code can be returned in multiple scenarios, not only when the contributors limit is hit. Please see how we handle the storage limit here:

https://github.com/MerginMaps/mobile/blob/404ac2b379289be2d1c5b387e1e2a11bc69dcdb2/core/merginapi.cpp#L2119-L2121

In fact, these two places should be unified 🙈

Reply:
Agreed it should be unified both of these place got advantage over the other

In the meantime I just added an extra safeguard in the sync error to parse the message for appropriate server error